python/GafferScene/ScriptProcedural.py

```diff
@@ -192,4 +202,48 @@ def __error( plug, source, error ) :
 			error
 		)
 
+	__performanceMonitor = None
+	@classmethod
+	def __ensurePerformanceMonitor( cls ) :
+
+		if cls.__performanceMonitor is not None :
+			return
+
+		cls.__performanceMonitor = Gaffer.PerformanceMonitor()
+		cls.__performanceMonitor.setActive( True )
+
+	@staticmethod
+	def __printItems( items ) :
+
+		if not len( items ) :
+			return
+
+		width = max( [ len( x[0] ) for x in items ] ) + 4
+		for name, value in items :
+			sys.stderr.write( "  {name:<{width}}{value}\n".format( name = name, width = width, value = value ) )
+
+	@classmethod
+	def __printStatisticsItems( cls, stats, key, n ) :
+
+		stats.sort( key = key, reverse = True )
+		items = [ ( x[0].relativeName( x[0].ancestor( Gaffer.ScriptNode ) ), key( x ) ) for x in stats[:n] ]
+		cls.__printItems( items )
+
+	@classmethod
+	def __printPerformance( cls ) :
+
+		## \todo This code is essentially just cribbed from the
```

Comment:
Any reason the monitor shouldn't provide its own "humanReadableStatistics" method that provides the nicely formatted dump of text for us? Or I guess a PerformanceMonitorUI helper class instead? Seems like the monitor implementer might have a decent idea of what the best layout for its statistics would be.